metadata.go

```diff
@@ -0,0 +1,52 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/h2oai/go-druid/builder/query"
+)
+
+type count struct {
+	Cnt int `json:"cnt"`
+}
+
+// MetadataService is a service that runs druid metadata requests using druid SQL API.
+type MetadataService struct {
+	client *Client
+}
+
+// AwaitDataSourceAvailable awaits for a datasource to be visible in druid table listing.
+func (md *MetadataService) AwaitDataSourceAvailable(dataSourceName string) error {
+	q := query.NewSQL()
+	q.SetQuery("SELECT count(*) cnt FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = 'druid' and TABLE_NAME='" + dataSourceName + "'")
```

Comment:
Using external data like is prone to SQL injection.

Please fix or provide a proof that data will always be sanitized or that that SQL injection is otherwise impossible.


`dataSourceName` name is part of the `WHERE` clause. Do we not support arguments/parameters for SQL?

Same for `AwaitRecordsCount`.